### PR TITLE
Document the library

### LINKED ISSUE #8: Create a microsite
Document and publish a microsite using [mdoc](https://github.com/scalameta/mdoc). Use ZIO's own documentation as reference.

### LINKED ISSUE #95: Design docs
Structure the notes gathered over Berlin hackathon.

### LINKED ISSUE #104: README Update
our README is outdated.

### PR DESCRIPTION
**Pull request by mijicd**
closes #8, closes #95, closes #104

### Summary

- Enable microsite and docs build and publish (thanks @softinio).
- Updated readme to resemble ZIO's.
- Created template docs consisted of overview, meetings and library sections.
- Extracted literature from readme to the separate section.

### Remarks

- Meetings shouldn't be left unmaintained.
- Proper design docs should be added under the `overview` section, as we reach the stable design. I didn't want to include any drafts at this point.
- Literature should be categorized and added over time.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # ZIO keeper
 
 [![CircleCI][badge-ci]][link-ci]
-[![Gitter][badge-gitter]][link-gitter]
 
 ZIO keeper is a purely-functional, type-safe library for managing distributed state in a strong
```
Comment (pshemass) [inline README.md]: I would definitely change that to reflect reality. Something like "ZIO Keeper is purely-functional set of libraries for distributed programming in Scala." then we can add later that we have core that provides transport and membership in the cluster and we are working on Leader Election, CRDT, routing, partitioning and sharding.

Comment (mijicd) [reply README.md]: Draft?:)

Comment (pshemass) [reply README.md]: I know I know just don't want to forget about this :)

File: README.md [hunk 2]
```diff
@@ -1,67 +1,35 @@
...
+It provides numerous primitives for tackling the common problems in distributed
+computing (e.g. leader election, cluster forming etc.). Under the hood, the
+library is backed by [ZIO][link-zio] and [ZIO NIO][link-nio], profiting from their
+performant, type and resource-safe APIs.
+
+- **Composable**. Design complex systems by composing the available building blocks.
+- **Resilient**. Build apps with automated failure recovery.
+- **Secure**. Benefit from security guarantees built into the library core.
+
+To learn more about ZIO keeper, check out the following references:
```
Comment (pshemass) [inline README.md]: this is perfect

File: docs/meetings/2019-11-22.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+---
+id: meetings_20191122
```
Comment (pshemass) [inline docs/meetings/2019-11-22.md]: I would just record meeting. there is no way that we gonna document them properly.

**Pull request merged**